appserver/payara-appserver-modules/microprofile/jwt-auth/src/main/java/fish/payara/microprofile/jwtauth/eesecurity/SignedJWTIdentityStore.java

```diff
@@ -0,0 +1,128 @@
+ * If you wish your version of this file to be governed by only the CDDL or
+ * only the GPL Version 2, indicate your decision by adding "[Contributor]
+ * elects to include this software in this distribution under the [CDDL or GPL
+ * Version 2] license."  If you don't indicate a single choice of license, a
+ * recipient has the option to distribute your version of this file under
+ * either the CDDL, the GPL Version 2 or to extend the choice of license to
+ * its licensees as provided above.  However, if you add GPL Version 2 code
+ * and therefore, elected the GPL Version 2 license, then the option applies
+ * only if the new code is made subject to such option by the copyright
+ * holder.
+ */
+package fish.payara.microprofile.jwtauth.eesecurity;
+
+import static java.lang.Thread.currentThread;
+import static javax.security.enterprise.identitystore.CredentialValidationResult.INVALID_RESULT;
+
+import java.io.IOException;
+import java.security.KeyFactory;
+import java.security.PublicKey;
+import java.security.spec.X509EncodedKeySpec;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Properties;
+
+import javax.security.enterprise.identitystore.CredentialValidationResult;
+import javax.security.enterprise.identitystore.IdentityStore;
+
+import fish.payara.microprofile.jwtauth.jwt.JsonWebTokenImpl;
+import fish.payara.microprofile.jwtauth.jwt.JwtTokenParser;
+
+/**
+ * Identity store capable of asserting that a signed JWT token is valid according to
+ * the MP-JWT 1.0 spec.
+ * 
+ * @author Arjan Tijms
+ */
+public class SignedJWTIdentityStore implements IdentityStore {
+    
+    private final JwtTokenParser jwtTokenParser = new JwtTokenParser();
+    
+    private String acceptedIssuer;
+    
+    public SignedJWTIdentityStore() {
+        try {
+            Properties properties = new Properties();
+            properties.load(currentThread().getContextClassLoader().getResource("/payara-mp-jwt.properties").openStream());
+            
+            acceptedIssuer = properties.getProperty("accepted.issuer");
+            
+        } catch (IOException e) {
+            throw new IllegalStateException("Failed to load properties", e);
+        }
+    }
+    
+    public CredentialValidationResult validate(SignedJWTCredential signedJWTCredential) {
+
+        try {
+            JsonWebTokenImpl jsonWebToken = 
+                    jwtTokenParser.parse(
+                            signedJWTCredential.getSignedJWT(), 
+                            acceptedIssuer, 
+                            readPublicKey("/publicKey.pem"));
+            
+            List<String> groups = new ArrayList<String>(
+                    jsonWebToken.getClaim("groups"));
+            
+            return new CredentialValidationResult(
+                    jsonWebToken, 
+                    new HashSet<>(groups));
+            
+        } catch (Exception e) {
+            // Ignore
```

Comment:
It could log at FINEST, sure.

The issue at hand is the TCK wants an exception here from the `jwtTokenParser` that's being used, plus that an exception can be thrown for all kinds of reasons where the token is invalid / corrupted / tampered with. You certainly don't want those exceptions in your normal log, as otherwise an attacker could all too easily DOS your server from log overflow by sending a stream of corrupted tokens. But FINE or so  should be uhm fine.